files/private-chef-cookbooks/private-chef/recipes/add_ons_local.rb

```diff
@@ -17,8 +17,19 @@
 
 addon_path = node['private_chef']['addons']['path']
 node['private_chef']['addons']['packages'].each do |pkg|
-  # find the newest package of each name
-  pkg_file = Dir["#{addon_path}/#{pkg}*.#{package_suffix}"].sort_by{ |f| File.mtime(f) }.last
+  if ::File.directory?(addon_path)
+    # find the newest package of each name if 'addon_path' is a directory
+    pkg_file = Dir["#{addon_path}/#{pkg}*.#{package_suffix}"].sort_by{ |f| File.mtime(f) }.last
+  else
+    if File.basename(addon_path) =~ /#{pkg}/
+      # use the full package path if it contains the package name
+      pkg_file = addon_path
+    else
```

Comment:
Thanks for the review @stevendanna.
The reason I'm verifying that 'pkg' is part of the file name is that node['private_chef']['addons']['packages'] comes as an array. Hence the original "node['private_chef']['addons']['packages'].each" loop.
Here's the output:

```
[root@ap-local-chef-server-12 chef]# chef-server-ctl install opscode-reporting --path /media/sf_mnt_kits/chef/opscode-reporting-1.3.0-1.el6.x86_64.rpm
---SNIP---
Recipe: private-chef::add_ons_local
  * yum_package[opscode-manage] action install (up to date)
  * yum_package[opscode-reporting] action install (up to date)
---SNIP---
Chef Client finished, 0/2 resources updated in 2.802826526 seconds
```

Without the parent directory check, it tries to install 'opscode-manage' using the full path of the reporting rpm.
I don't know why we have an array of packages. Is it an attribute cache/merging issue? I've also noticed  "node['private_chef']['addons']['path']" getting stuck to '/media/sf_mnt_kits/chef/' even if I run the install without --path, like this:

```
[root@ap-local-chef-server-12 chef]# chef-server-ctl install chef-sync
---SNIP---

From: /opt/opscode/embedded/cookbooks/cache/cookbooks/private-chef/recipes/add_ons_local.rb @ line 19 Chef::Mixin::FromFile#from_file:

    18: require "pry"
 => 19: binding.pry
    20:
    21: addon_path = node['private_chef']['addons']['path']
    22:
    23: node['private_chef']['addons']['packages'].each do |pkg|

[1] pry(#<Chef::Recipe>)> node['private_chef']['addons']['path']
=> "/media/sf_mnt_kits/chef/opscode-reporting-1.3.0-1.el6.x86_64.rpm"
[2] pry(#<Chef::Recipe>)> node['private_chef']['addons']['packages']
=> ["opscode-manage", "opscode-reporting", "chef-sync"]
[3] pry(#<Chef::Recipe>)> quit
Converging 6 resources
Recipe: private-chef::add_ons_wrapper
  * ruby_block[addon_install_notification_opscode-manage] action nothing (skipped due to action :nothing)
  * ruby_block[addon_install_notification_opscode-reporting] action nothing (skipped due to action :nothing)
  * ruby_block[addon_install_notification_chef-sync] action nothing (skipped due to action :nothing)
Recipe: private-chef::add_ons_local
  * yum_package[opscode-manage] action install (up to date)
  * yum_package[opscode-reporting] action install (up to date)
  * yum_package[chef-sync] action install
    * Package chef-sync not found:
---SNIP---
[2015-04-08T14:33:21+01:00] FATAL: Chef::Exceptions::Package: yum_package[chef-sync] (private-chef::add_ons_local line 38) had an error: Chef::Exceptions::Package: Package chef-sync not found:
```

I've looked into the attributes file used by the chef-client run:
```
/opt/opscode/embedded/bin/ruby /opt/opscode/embedded/bin/chef-client -z --config /opt/opscode/embedded/cookbooks/solo.rb --json-attributes /opt/opscode/embedded/cookbooks/install_params.json --log_level fatal
```
and the path should be null:
```
[root@ap-local-chef-server-12 ~]# cat /opt/opscode/embedded/cookbooks/install_params.json
{"run_list":["recipe[private-chef::add_ons_wrapper]"],"private_chef":{"addons":{"install":true,"packages":["chef-sync"],"path":null}}}
```